src/Microsoft.AspNet.Mvc.ViewFeatures/ViewFeatures/DefaultHtmlGenerator.cs

```diff
@@ -800,8 +800,10 @@ public virtual IHtmlContent GenerateAntiforgery(ViewContext viewContext)
             var htmlSummary = new TagBuilder("ul");
             foreach (var modelState in modelStates)
             {
-                foreach (var modelError in modelState.Errors)
+                // Perf: Avoid allocations
+                for (var i = 0; i < modelState.Errors.Count; i++)
```

Comment:
This one shouldn't be a hot path. Never have more than one validation summary in a page.